dotnet-engine/Yggdrasil.Engine/Types.cs

```diff
@@ -72,6 +72,21 @@ public override int GetHashCode()
     }
 }
 
+public class FeatureDefinition
+{
+    public FeatureDefinition(string name, string project, string? type)
+    {
+        Name = name;
+        Project = project;
+        Type = type;
+    }
+    public string Name { get; set; }
+
+    public string Project { get; set; }
+
+    public string? Type { get; set; }
```

Comment:
Is the type optional?

Reply:
Yeeeaa... maybe. This is a good question. I've derived the type here from the types defined in our core Unleash Types rust library, which says this is optional. As for the actual question. I don't think this is optional in modern versions of Unleash but it's non existent in older versions. Not sure how far we back we have to go but there's at least one set of humans using Unleash v3 and I just haven't checked